dss/util/auth/authorize.py

```diff
@@ -1,28 +1,52 @@
 import typing
+import logging
 
-from dss.util.auth import authregistry
+from dss.config import Config
+from dss.error import DSSForbiddenException, DSSException
+from .authregistry import AuthRegistry
 
+logger = logging.getLogger(__name__)
 
-class Authorize(metaclass=authregistry.AuthRegistry):
-    """ abstract class for authorization classes"""
+
+class Authorize(metaclass=AuthRegistry):
+    """
+    Base class for authentication/authorization methods.
+    This class exists solely to wrap the definition of the assert security flow.
+    Subclasses must define a security_flow method.
+    All subclasses inherit from AuthRegistry and are added to a registry on definition.
+    """
     def __init__(self):
         pass
 
-    def security_flow(self, authz_methods: typing.List[str], *args, **kwargs):
+    def security_flow(self, *args, **kwargs):
         """
         This function maps out flow for a given security config
+        Be sure to call it from the derived class to ensure the JWT has groups set
         """
-
-        raise NotImplementedError()
+        groups = kwargs.get('security_groups')
```

Comment:
This should use the `assert_required_parameters()` method for both `groups` and `token`. Also does the `token` refer to the JWT?

Reply:
Wait - just realized you're already handling the case of missing parameters. This method should be simplified by just using `assert_required_parameters()` and then call `assert_authorized_group()` rather than handling the case of one or the other being None

Reply:
wowow cant believe i forgot about that function, and yes!

Reply:
addressed :)